app/ts/components/IPFSSubpathRedirect.tsx

```diff
@@ -0,0 +1,31 @@
+import { CID } from 'multiformats'
+import { useEffect } from 'preact/hooks'
+
+export const IPFSSubpathRedirect = () => {
+	function extractHashFromBasePath() {
+		const htmlBase = document.querySelector('base')
+		if(!htmlBase) return
+		const cidRegex = /\/ipfs\/(Qm[1-9A-HJ-NP-Za-km-z]{44,}|b[A-Za-z2-7]{58,}|B[A-Z2-7]{58,}|z[1-9A-HJ-NP-Za-km-z]{48,}|F[0-9A-F]{50,})/
```

Comment:
Did you come up with this, or did you find it online somewhere?

Reply:
Is there no function in multiformats like `isCid`?

Reply:
> Did you come up with this, or did you find it online somewhere?

https://stackoverflow.com/questions/67176725/a-regex-json-schema-pattern-for-an-ipfs-cid
it was linked from some writing/blog but couldn't trace back which one i got it from 😖

Reply:
Consider using https://multiformats.github.io/js-multiformats/classes/cid.CID.html#asCID-2 instead.  According to docs, it will return null if the input value is not a CID.

Something like this (pseudocode, untested):
```ts
if !(path.contains('/ipfs/')) return
const maybeCidMatch = /\/ipfs\/(.*?)\//.exec(path)
if (!maybeCidMatch) return
const cid = CID.asCID(maybeCidMatch[1]!)
return cid
```

Reply:
This part of the code only extracts the hash/cid from base element that should be parsed by another function so simplified the regex match

```ts
const htmlBase = document.querySelector('base')
if (!htmlBase || !htmlBase.href.includes('/ipfs/')) return
const maybeCidMatch = /\/ipfs\/(.*?)\//.exec(htmlBase.href)
return maybeCidMatch?.at(1)
```